### PR TITLE
8260684: vmTestbase/gc/gctests/PhantomReference/phantom002/TestDescription.java timed out

### PR DESCRIPTION
**Pull request by kimbarrett**
Please review this fix of
vmTestbase/gc/gctests/PhantomReference/phantom001/phantom001.java to
eliminate races that can lead to the test occasionally timing out.

When the referent being tested by a particular iteration is a finalizable
class, the test invoked eatMemory() repeatedly until the referent's
finalize() function set a flag.  Then, whether the referent is finalizable
or not, there is a call to eatMemory() to cause the phantom reference to be
cleared and notified.  The testing thread then proceeds to wait for the
reference to be notified.

This has a problem with ZGC.  One collection made the referent finalizable.
Reference processing and finalization will end up ensuring it will survive
the following collection.  If the post-finalization eatMemory() is that next
collection, then it won't clear and notify the associated PhantomReference.
Usually that doesn't cause a problem because the referent will be reclaimed
by some other thread's later call to eatMemory().  But if the test is done
and terminating, there won't be a later GC, and the wait for notification
will block until the test times out.

There are other, rarer, scenarios with various collectors that could lead to
similar timeouts, but the one described above is "relatively" likely.

While investigating this issue I did a bunch of code cleanup (for example,
there was some really badly formatted code), refactoring, and commenting.
As a result, a large fraction of the test has changed, though mostly in ways
that don't affect it's function.  The first of the two commits in the PR is
a whitespace-only cleanup.  Excluding it may make reviewing easier.

Testing:
Ran the phantom002 test a couple thousand times.

Using a modified version of the test, verified that the described problem
case actually occurs, on the order of 1% of the time.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8260684](https://bugs.openjdk.java.net/browse/JDK-8260684): vmTestbase/gc/gctests/PhantomReference/phantom002/TestDescription.java timed out

### Reviewers
 * [Per Liden](https://openjdk.java.net/census#pliden) (@pliden - **Reviewer**) ⚠️ Review applies to 5238becabaed1bf2aa574d45a367cbfcd187b03a
 * [Leo Korinth](https://openjdk.java.net/census#lkorinth) (@lkorinth - Committer) ⚠️ Review applies to 5238becabaed1bf2aa574d45a367cbfcd187b03a

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk17 pull/142/head:pull/142` \
`$ git checkout pull/142`

Update a local copy of the PR: \
`$ git checkout pull/142` \
`$ git pull https://git.openjdk.java.net/jdk17 pull/142/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 142`

View PR using the GUI difftool: \
`$ git pr show -t 142`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk17/pull/142.diff">https://git.openjdk.java.net/jdk17/pull/142.diff</a>

</details>

### CODE REVIEW

**Review by pliden — APPROVED** [review]
Looks good.

**Review by lkorinth — APPROVED** [review]
Looks good to me!

File: test/hotspot/jtreg/vmTestbase/gc/gctests/PhantomReference/phantom001/phantom001.java [hunk 1]
```diff
@@ -98,12 +99,43 @@ public static void main(String[] args) {
...
+        }
+
+        private boolean shouldTerminate() {
+            return !getExecutionController().continueExecution();
+        }
+
+        private void eatMemory(int initialFactor) {
+            GarbageUtils.eatMemory(getExecutionController(),
+                                   garbageProducer,
+                                   initialFactor , 10, 0);
```
[the remark this answers is not in the document]
Comment (kimbarrett) [reply test/hotspot/jtreg/vmTestbase/gc/gctests/PhantomReference/phantom001/phantom001.java]: Done.

**Comment by kimbarrett** [comment]
Thanks for reviews @pliden and @lkorinth 

/integrate

**Pull request closed without merging**